persistent/Database/Persist/Class/PersistEntity.hs

```diff
@@ -452,7 +451,7 @@ idField = "_id"
 --   toPersistValue = toPersistValueJSON
 -- @
 toPersistValueJSON :: ToJSON a => a -> PersistValue
-toPersistValueJSON = PersistText . LT.toStrict . TB.toLazyText . encodeToTextBuilder . toJSON
+toPersistValueJSON = PersistByteString . B.toStrict . encodingToLazyByteString . toEncoding
```

Comment:
I'm a little concerned that changing the `PersistValue` constructor here will cause downstream issues for consumers, since the constructor tag is used to determine things like which postgres type is used.

ie anyone who is expecting for `case toPersistValueJSON x of PersistText x -> ...` is going to be broken (which would be *odd*, of course)

It seems like we can do this in a safer way by just dropping the `. toJSON` in the original?

```suggestion
toPersistValueJSON = PersistText . LT.toStrict . TB.toLazyText . encodeToTextBuilder
```

since [`encodeToTextBuilder`](https://www.stackage.org/haddock/lts-22.28/aeson-2.1.2.1/Data-Aeson-Text.html#v:encodeToTextBuilder) is `ToJSON a => a -> Builder` already.

Though the docs for that says: "Uses `toJSON`", and immediately above, we have [`encodeToLazyText`](https://www.stackage.org/haddock/lts-22.28/aeson-2.1.2.1/Data-Aeson-Text.html#v:encodeToLazyText), which changes this to:

```suggestion
toPersistValueJSON = PersistText . LT.toStrict . encodeToLazyText
```

Reply:
yeah I saw that encodeToTextBuilder uses toJSON, but didn't realize encodeToLazyText uses toEncoding

Reply:
Yeah PersistText seems fine. I think it's a little extra work though - it has to decode utf8 then encode it again. I think there's some fusion things GHC has that may mean it doesn't actually do this internally though.